### PR TITLE
register fluid kerenls to phi [part 6]

### PR DESCRIPTION
**Pull request by huangjiyi**
### PR types

Others

### PR changes

Others

### Describe

Pcard-67000

Register fluid kernels of 39 ops to phi:
- gaussian_random
- generate_mask_labels
- generate_proposal_labels
- get_float_status
- get_tensor_from_selected_rows
- global_gather
- global_scatter
- graph_khop_sampler
- group_norm
- gru
- gru_unit
- hash
- hinge_loss
- im2sequence
- inplace_abn
- iou_similarity
- l1_norm
- lars_momentum
- limit_by_capacity
- linear_chain_crf
- load_combine
- locality_aware_nms
- lod_reset
- lookup_table
- lookup_table_dequant
- lrn
- lstm
- lstmp
- lstm_unit
- margin_rank_loss
- marker
- match_matrix_tensor
- mean_iou
- mine_hard_examples
- minus
- modified_huber_loss
- mp_allreduce_sum
- multiclass_nms
- multihead_matmul

### CODE REVIEW

File: paddle/fluid/operators/mean_iou_op.cc [hunk 1]
```diff
@@ -107,7 +107,6 @@ REGISTER_OPERATOR(
     ops::MeanIoUOpMaker,
     paddle::framework::EmptyGradOpMaker<paddle::framework::OpDesc>,
     paddle::framework::EmptyGradOpMaker<paddle::imperative::OpBase>);
-REGISTER_OP_CPU_KERNEL(mean_iou,
-                       ops::MeanIoUKernel<int>,
-                       ops::MeanIoUKernel<int32_t>,
-                       ops::MeanIoUKernel<int64_t>);
+
+PD_REGISTER_STRUCT_KERNEL(
+    mean_iou, CPU, ALL_LAYOUT, ops::MeanIoUKernel, int, int64_t) {}
```
Comment (YuanRisheng) [inline paddle/fluid/operators/mean_iou_op.cc]: 这里漏了int32_t

File: paddle/fluid/operators/mean_iou_op.cu [hunk 2]
```diff
@@ -166,7 +166,5 @@ class MeanIoUCUDAOpKernel : public framework::OpKernel<T> {
 }  // namespace paddle
 
 namespace ops = paddle::operators;
-REGISTER_OP_CUDA_KERNEL(mean_iou,
-                        ops::MeanIoUCUDAOpKernel<int>,
-                        ops::MeanIoUCUDAOpKernel<int64_t>,
-                        ops::MeanIoUCUDAOpKernel<int32_t>);
+PD_REGISTER_STRUCT_KERNEL(
+    mean_iou, GPU, ALL_LAYOUT, ops::MeanIoUCUDAOpKernel, int, int64_t) {}
```
Comment (YuanRisheng) [inline paddle/fluid/operators/mean_iou_op.cu]: 同上

File: paddle/fluid/operators/mean_iou_op.cc (hunk 1 above)
Comment (huangjiyi) [reply paddle/fluid/operators/mean_iou_op.cc]: done

File: paddle/fluid/operators/mean_iou_op.cu (hunk 2 above)
Comment (huangjiyi) [reply paddle/fluid/operators/mean_iou_op.cu]: done

**Pull request closed without merging**